.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,33 +1,164 @@
+- [ ] 📝 I have commented my code, especially in hard-to-understand areas (e.g., using JSDoc for methods).
+- [ ] ⚠️ My changes generate no new warnings.
+- [ ] 🤖 My code needed automated testing. I have added them (this is an optional task).
+- [ ] 📄 Documentation updates are included (if applicable).
+- [ ] 🔒 I have considered potential security impacts and mitigated risks.
+- [ ] 🧰 Dependency updates are listed and explained.
+- [ ] 📚 I have read and understood the [Pull Request guidelines](../CONTRIBUTING.md#recommended-pull-request-guideline).
+
+## 📷 Screenshots or Visual Changes
+
+Please upload the image directly here by pasting it or dragging and dropping. Avoid using external image services as the image will be uploaded automatically.
+
+<details><summary>Click here for more details:</summary>
+</p>
+
+If this pull request introduces visual changes, please provide the following details:
+
+- **UI Modifications**: Highlight any changes made to the user interface.
+- **Before & After**: Include screenshots or comparisons (if applicable).
+
+| Event              | Before                | After                |
+| ------------------ | --------------------- | -------------------- |
+| `UP`               | ![Before](image-link) | ![After](image-link) |
+| `DOWN`             | ![Before](image-link) | ![After](image-link) |
+| Certificate-expiry | ![Before](image-link) | ![After](image-link) |
+| Testing            | ![Before](image-link) | ![After](image-link) |
+
+</p>
+</details>
+
+## ℹ️ Additional Context
+
+Provide any relevant details to assist reviewers in understanding the changes.
+
+<details><summary>Click here for more details:</summary>
+</p>
+
+**Key Considerations**:
+
+- **Design decisions** – Key choices or trade-offs made during development.
+- **Alternative solutions** – Approaches considered but not implemented, along with reasons.
+- **Relevant links** – Specifications, discussions, or resources that provide context.
+- **Dependencies** – Related pull requests or issues that must be resolved before merging.
+- **Additional context** – Any other details that may help reviewers understand the changes.
+
+Provide details here
+
+## 📄 Document Status
+
+To streamline the review process, please list the filenames of documents currently being worked on. This helps reviewers focus on the files that need their attention.
+
+### ✅ Ready for Review
+
+The following documents are complete and ready for review:
+
+- `Specify filenames here`
+
+### 🔨 Work in Progress
+
+The following documents are currently being worked on:
+
+- `List ongoing documents here`
+
+### 💬 Requested Feedback
+
+Your feedback is requested on the following documents:
+
+- `Mention documents needing feedback here`
+
+## 📝 Additional Changes
+
+Please provide any relevant details or context. (e.g., resolved Linter issues, code optimizations, or other adjustments):
+
+- Describe additional changes here
+
+</p>
+</details>
```

Comment:
Unsure what we would recive here. If signal to noice is too low, nobody is reading this => lets remove it

```suggestion
```